src/encoded/tests/test_search.py

```diff
@@ -1,12 +1,170 @@
 # Use workbook fixture from BDD tests (including elasticsearch)
-from .features.conftest import app_settings, app, workbook
+from collections import OrderedDict
 from webob.multidict import MultiDict
 import pytest
+from encoded.tests.features.conftest import app_settings, app, workbook
+from snovault import TYPES
+from snovault.helpers.helper import (
+    set_filters,
+    set_facets,
+    format_facets,
+    sort_query,
+    get_pagination,
+    prepare_search_term,
+    set_sort_order,
+    get_search_fields,
+    list_visible_columns_for_schemas,
+    list_result_fields,
+    build_terms_filter,
+    build_aggregation,
+    normalize_query
+)
 
 
-# Integration tests
+class FakeRequest(object):
+    path = '/search/'
+
+    def __init__(self, params):
+        self.params = MultiDict(params)
+        self.permissions = {}
+
+    def set_permission(self, perm_key, perm_value=False):
+        self.permissions[perm_key] = perm_value
+
+    def has_permission(self, perm_key):
+        if perm_key in self.permissions and self.permissions[perm_key] is True:
+            return True
+        return False
+
+
+@pytest.fixture
+def sort_query_item():
+    item = [({'d': 'val', 'c': 'val',
+              'b': 'val', 'a': 'val'},
+             OrderedDict([('a', 'val'), ('b', 'val'),
+                          ('c', 'val'), ('d', 'val')])),
+            ({'d': {'b': 'val', 'a': 'val'}},
+             OrderedDict([('d', OrderedDict([('a', 'val'),
+                                             ('b', 'val')]))]))]
+    return item
+
+
+@pytest.fixture
+def get_pagination_item():
+    requests = {FakeRequest((('from', '0'), ('limit', '25'))): ('0', 25),
+                FakeRequest((('from', '10'), ('limit', '50'))): ('10', 50),
+                FakeRequest((('from', '0'),)): ('0', 25),
+                FakeRequest((('limit', '50'),)): (0, 50),
+                FakeRequest((('key', 'value'),)): (0, 25)}
+    return requests
+
+
+@pytest.fixture
+def prepare_search_term_item():
+    requests = {FakeRequest((('type', 'Page'),
```

Comment:
This first test doesn't need all these extra parameters.  It just needs to not contain 'searchTerm'.   So no args are just as good as these four.  Also why not group the ones that return early with '*' together.  May want to add a third one where the FakeRequest has searchTerm = '*'.


Also, without good reason let's not return a dictionary with key values pairs when we can be consistent and return lists of sets with the test object and test results.